src/common/snippets/include/snippets/lowered/pass/pass.hpp

```diff
@@ -49,20 +50,58 @@ class Pass {
 
 class PassPipeline {
 public:
-    PassPipeline() = default;
+    PassPipeline();
+    PassPipeline(const std::shared_ptr<PassConfig>& pass_config);
 
+    class PassPosition {
+    public:
+        enum class Place {Before, After, PipelineStart, PipelineEnd};
+        explicit PassPosition(Place pass_place);
+        explicit PassPosition(Place pass_place, const DiscreteTypeInfo& pass_type_info, size_t pass_instance = 0);
+
+        std::vector<std::shared_ptr<Pass>>::const_iterator get_insert_position(const std::vector<std::shared_ptr<Pass>>& pass_list) const;
+
+    private:
+        const DiscreteTypeInfo m_pass_type_info = {};
+        const size_t m_pass_instance = 0;
+        const Place m_place = Place::Before;
+    };
```

Comment:
Just an idea: can we move PassPosition class to a separate file and reuse it in `snippets::Manager` and `PassPipeline`? It seems like we can easily do that. `get_insert_position` in this case will become template method

Reply:
Good idea. Moreover, I create template class `PositionedPass`. Thanks for the idea!